src/backend/storage/file/buffile.c

```diff
@@ -998,6 +1000,18 @@ BufFilePledgeSequential(BufFile *buffile)
 
 #define BUFFILE_ZSTD_COMPRESSION_LEVEL 1
 
+void *
+customAlloc(void* opaque, size_t size)
+{
+	return gp_malloc(size);
```

Comment:
why not palloc?

Reply:
gp_malloc and gp_free keep track of virtual memory which is the goal of the patch
https://github.com/arenadata/gpdb/blob/2de3b73a03045062849da9b99612bcd022334a7b/src/backend/utils/mmgr/memprot.c#L443

Reply:
But palloc is a wrapper over gp_malloc.

Reply:
Haven't found this definition, only that one which is a wrapper to pg_malloc, can you point to it?

Reply:
> can you point to it?

https://github.com/arenadata/gpdb/blob/2de3b73a03045062849da9b99612bcd022334a7b/src/backend/utils/mmgr/mcxt.c#L1297-L1314

Reply:
Seems like it uses `MemoryContextAlloc`, do you mean that it calls `gp_malloc` internally?

Reply:
> Seems like it uses `MemoryContextAlloc`, do you mean that it calls `gp_malloc` internally?

yes